core/src/main/kotlin/org/eln2/data/MultiMap.kt

```diff
@@ -0,0 +1,197 @@
+package org.eln2.data
+
+/**
+ * A helper class for representing a mapping as a Map.Entry.
+ */
+data class MultiMapEntry<K, V>(override val key: K, override val value: V): Map.Entry<K, V>
+
+/**
+ * A MultiMap: a data structure that maps one key (K) to zero or more values (V).
+ *
+ * This is in contrast to the usual Map, which maps one key to at most one value.
+ *
+ */
+interface MultiMap<K, V> {
+	/**
+	 * Tests whether a key [k] is in the MultiMap.
+	 *
+	 * This is true if k is associated with at least one value.
+	 */
+	operator fun contains(k: K): Boolean = get(k).isNotEmpty()
+
+	/**
+	 * Get the set of all values associated with key [k].
+	 */
+	operator fun get(k: K): Set<V>
+
+	/**
+	 * An iterator over every Entry<K, Set<V>>, of size keySize.
+	 */
+	val keyMapping: Iterable<Map.Entry<K, Set<V>>>
+	/**
+	 * The number of keys, value sets, and keyMapping entries in this MultiMap.
+	 */
+	val keyMappingSize: Int get() = keyMapping.map { 1 }.sum()
```

Comment:
Could you use .count() instead?

I know it's an iterable, but the underlying data structure isn't.

Reply:
Didn't know about count! Will do once I get back to an IDE :)